app/drivers/usb/cdc.c

```diff
@@ -221,8 +228,7 @@ err_t usb_cdc_tx(uint8_t *p_buf, uint16_t len)
 	if (self.p_usbd->dev_state != USBD_STATE_CONFIGURED)
 		return EUSB_CDC_NOT_READY;
 
-	if (self.tx_busy)
-		return EUSB_CDC_BUSY;
+	xSemaphoreTake(self.tx_done_semaphore, portMAX_DELAY);
 
 	self.tx_busy = true;
```

Comment:
the tx_busy flag can be remove then? Seems to double the semaphore's functionality.

Reply:
true